### PR TITLE
UI for file inclusion

### PR DESCRIPTION
**Pull request by scottmmjackson**
### Description

- Add a bootstrap-based UI that allows adding and removing files from the `files` list.
- ~~Add a button that adds all files.~~
- Add the files parameter to the publish function
- Changes the default app mode to rendered and not static.
- **NEW IN RECENT COMMIT**: Default server configurations with no `disableTLSCheck` flag to `false`.

I ended up preserving the full path through all of the arguments so that we can remove the base path at the last possible opportunity. We can decide whether we want to do that in JS or Python as a follow-on commit to the last validation of `mm-files-in-bundle`.

### Testing Notes / Validation Steps

- Adding/removing files should work for everything except the base notebook, its manifest, and its `requirements.txt`
- Test that you can add/remove files from the file dialog
- Test that you can remove files from the publish dialog
- When you redeploy content, we destroy the publish dialog, open a dialog that has you select either a new deployment or an existing deployment, and then recreate the publish dialog. **Test that the deployed files are consistent with the file list in this case as well**

### CODE REVIEW

File: rsconnect_jupyter/static/connect.js [hunk 1]
```diff
@@ -147,6 +151,121 @@ define([
...
+      updateListGroupItems: function() {
+        var that = this;
+        that.$listGroup.empty();
+        that.contents.forEach(function (item) {
+          if (that.fileList.indexOf(item.path) !== -1) {
+            var li = document.createElement('li');
+            var i = document.createElement('i');
+            i.className = 'fa fa-times';
+            li.className = 'list-group-item';
+            li.appendChild(i);
```
Comment (mmarchetti) [inline rsconnect_jupyter/static/connect.js]: This is fine as is, I'm just curious why you didn't create the elements with jQuery?

File: rsconnect_jupyter/static/connect.js [hunk 2]
```diff
@@ -706,14 +825,18 @@ define([
...
+        '          <button class="btn btn-default dropdown-toggle" type="button" id="file-select" data-toggle="dropdown" disabled>',
+        '          Add a file to deploy',
+        '          <span class="caret"></span>',
+        '          </button>',
+        '          <ul class="dropdown-menu" aria-labeledby="file-select" id="file-select-menu">',
+        '          </ul>',
+        '        </div>',
+        '        <ul class="list-group" id="file-list-group">',
+        '        </ul>',
+        '        <button class="btn btn-default" type="button" id="file-include-all">Add All Files In Directory</button>',
```
Comment (mmarchetti) [inline rsconnect_jupyter/static/connect.js]: Can this button be on the same line as the "Add a File" button?

**Comment by mmarchetti** [comment]
I test this locally. It worked well, but I ran into an issue when I created a subdirectory at the same level as the notebook file.

```
{…}
abort: function abort()​
always: function always()​
catch: function catch()​
done: function add()​
fail: function add()​
getAllResponseHeaders: function getAllResponseHeaders()​
getResponseHeader: function getResponseHeader()​
overrideMimeType: function overrideMimeType()​
pipe: function pipe()​
progress: function add()​
promise: function promise()
readyState: 4
responseJSON: Object { message: "No such file or directory: Basic Numpy Notebook.ipyn", reason: null }
responseText: "{\"message\": \"No such file or directory: Basic Numpy Notebook.ipyn\", \"reason\": null}"
setRequestHeader: function setRequestHeader()​
state: function state()
status: 404
statusCode: function statusCode()
statusText: "Not Found"
then: function then()​
<prototype>: Object { … }
utils.js:731
API request failed (404): No such file or directory: Basic Numpy Notebook.ipyn 

Error: No such file or directory: Basic Numpy Notebook.ipyn 

TypeError: that.fileListItemManager is null
    open connect.js:944
```

File: rsconnect_jupyter/static/connect.js (hunk 1 above)
Comment (scottmmjackson) [reply rsconnect_jupyter/static/connect.js]: It's a habit I've gotten into that, in some instances, helps to minimize browser HTML parsing passes. It's kind of an overoptimization unless you do it by habit.

**Comment by scottmmjackson** [comment]
Ahhh, it looks like I have to unroll subdirectories...

**Comment by scottmmjackson** [comment]
Unless we provide the subdirectories as selectable options that can be traversed in their entirety if included.

@slopp What do you think about this? I can highlight when entire directories are being included in the UI. I feel like that would make things a bit easier.

**Comment by scottmmjackson** [comment]
Per slack I'm merging this into the branch and then preparing the `mm-files-in-bundle` branch for review.

**Pull request merged**